src/Providers/EntityManagerProvider.php

```diff
@@ -16,46 +16,42 @@
 class EntityManagerProvider extends AbstractServiceProvider
 {
     protected $provides = [
-        ORM\EntityManager::class
+        ORM\EntityManagerInterface::class
     ];
 
     public function register()
     {
-        $this->container->share(
```

Comment:
This is probably an over optimization, but if I remember correctly, the callback is only called when a request for the class in question is made and it needs to be instantiated. Until then, it just stores the closure waiting to be called. 

Whereas, in your current implementation, the code is all executed on `register`. 

So, slightly more performant the other way around?